src/blockdata/opcodes.rs

```diff
@@ -786,7 +786,21 @@ pub enum Class {
     Ordinary(Ordinary)
 }
 
-display_from_debug!(Class);
+impl fmt::Display for  Class {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Class::PushNum(ref number ) => write!(f, "Push number {}", number),
+            Class::PushBytes(ref bytes ) => write!(f, "Push bytes {}", bytes),
+            Class::ReturnOp => write!(f, "ReturnOp"),
+            Class::SuccessOp => write!(f, "SuccessOp"),
+            Class::IllegalOp => write!(f, "IllegalOp"),
+            Class::NoOp => write!(f, "NoOp"),
+            Class::Ordinary(ordinary) => write!(f, "Ordinary:  {}", ordinary),
```

Comment:
I think we should avoid spaces like this, and maybe use `Ordinary({})` or `{{ ordinary: {} }}`.

Reply:
Should this one even have `Display`?

Reply:
Good question. Probably not actually -- its meaning is pretty specific to this library.

Reply:
True, we don't need `Display` but we need `Debug`. I removed implementation and added `Debug` in the `derive` attribute

Reply:
I spoke too soon, we actually need `Display` for serialization

Reply:
Do we need this to be serde serializable? :)

Reply:
And even if yes, why would we use this format?! Either `#[derive(Serialize)]` or specify something that's not based on debug representation.